app/views/admin_applications/show.html.erb

```diff
@@ -0,0 +1,18 @@
+<h2 id="header">
+  <%= @application.name %></br>
+  <%= @application.status %>
+</h2>
+
+<% @application.pet_applications.each do |pet_application| %>
+  <section id="pet<%= pet_application.pet_id %>">
+    <%= pet_application.pet.name %>
+    <% if pet_application.approved %>
+      </br>This pet is approved
+    <% elsif pet_application.rejected %>
+      </br>This pet is rejected
+    <% else %>
+      <%= button_to "Approve this Pet", "/admin/applications/#{@application.id}", method: :patch, params: { approved: true, pet_id: pet_application.pet_id }, data: { turbo: false } %>
```

Comment:
What we're doing here is actually updating the joins record that is associating 2 existing resources: a pet and an application. So, it's best to have our route reflect these 2 resources. The 2 most common RESTful paths include:
"/applications/:app_id/pets/:pet_id" or "application_pets/:id" where this last ID is the primary key for the joins table record associating these 2

Note that neither of these have `admin` prepended to them, because this route is triggered from a button rather than a URI that is entered by the user, so it's sort of an internal route rather than external. We could definitely keep `admin` but it's not strictly necessary. So it could route to `pet_applications#update`

Reply:
Also, Turbo shouldn't interfere with buttons so you should be able to remove the `data: { turbo: false }`